### ISSUE TITLE
Same script, different performance with xgboost on two enviornments (Windows Laptop v.s. Kaggle Script Env)

### ISSUE THREAD

**Issue by yifanxie** [dmlc/xgboost#391]
Firstly, allow me to say a huge thanks for your work to produce this extreme useful package. 
It is my first post here, and very much new to xgboost, so please bear with me.

I am participating the Caterpillar Tube Pricing challenge, and meet somehow a puzzling question: I am running the same scrip using xgboost under firstly my own laptop (windows 8, winpython 3.4)  and then secondly under Kaggle's scripting environment ( python 3.4.3, Anaconda 2.2.0 (64-bit)), and the performance on Kaggle's environment somehow is always better?

My cross validation script is [here](https://www.kaggle.com/yifanxie/caterpillar-tube-pricing/training-set-split-for-cross-validation), and the performance of RMSE based on 10 performance run is the following:

- Kaggle script environment:  0.214990
- My laptop: 0.216969

Basically the there is a good 0.002 RMSE difference!
I have also run the same basic script to generate the submission file, and again, always better result from the Kaggle scripting environment. 

It seems to only apply to xgboost,  I have compare the results of my script using sklearn linear regression and random forest, there results are more or less the same in the two environments.

Has anyone experience the same issue? and how do you solve it?  is it simply an Window issue or is it to do with the optimization tool used by xgboost under different setting?

Best Regards
Yifan

**Comment by tqchen** [comment]
Should be related to #310

**Comment by yifanxie** [comment]
@tqchen thank you very much for getting back to me! I will have a look at the post you refer to, and come back to report for any progress :) :) :)

Best Regards
Yifan

**Comment by mirilias** [comment]
@yifanxie hello, i have the same issue, can you tell me which "seed" value did you explicit

**Comment by tqchen** [comment]
random number generator is unified in #735
